src/brushes/base_brush.class.js

```diff
@@ -68,9 +68,10 @@ fabric.BaseBrush = fabric.util.createClass(/** @lends fabric.BaseBrush.prototype
   /**
    * Sets brush styles
    * @private
+   * @param {CanvasRenderingContext2D} [ctx]
    */
-  _setBrushStyles: function() {
-    var ctx = this.canvas.contextTop;
+  _setBrushStyles: function (ctx) {
+    ctx = ctx || this.canvas.contextTop;
```

Comment:
we are doing for 5.0 and setBrush style is internal, it is fine to check where this is called all over fabric an just update it to be used with the ctx parameter.

Reply:
3ad6470d